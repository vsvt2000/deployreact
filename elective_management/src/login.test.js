import React from 'react';
import { shallow } from 'enzyme';
import Login from './App';
import {mount} from 'enzyme';
import axios from 'axios';
    describe('Login component tests', ()=> {
        const wrapper = shallow(<Login />);

        it('should have a btn component', ()=>{
            //There should be only one button
            expect(wrapper.find('button#Button')).toHaveLength(1);


            //Button should have matching text
            expect(JSON.parse(JSON.stringify(wrapper.find('button').text()))).toEqual("Sign In");
        });

        it('should have the Amrita Logo', ()=> {

            //There should be only one button
            expect(wrapper.find('img#Amrita')).toHaveLength(1);
            expect(wrapper.find('div#data')).toHaveLength(1);

        });

        it('should have input for email and password', ()=> {
            //Email and password input field should be present
            expect(wrapper.find('input#email')).toHaveLength(1);
            expect(wrapper.find('input#pwd')).toHaveLength(1);
        });


        it('should have an empty email and password state var', ()=> {
            expect(wrapper.state('email')).toEqual('');
            expect(wrapper.state('password')).toEqual('');
        });

        it('simulating value entries',()=>{
            const credentials={
                username:"cb.en.u4cse18042@cb.students.amrita.edu",
                password:"logind"
            }
            const usernameInput = wrapper.find('input#email');
            usernameInput.value = credentials.username;
            expect(usernameInput.value).toBe('cb.en.u4cse18042@cb.students.amrita.edu');

            const passwordInput = wrapper.find('input#pwd');
            passwordInput.value = credentials.password;
            expect(passwordInput.value).toBe('logind');
            wrapper.find('button#Button').simulate('click');
            expect(wrapper.find('h5#fail')).toEqual({});
        })

        it('makes axios post request', ()=> {
            let request={
                email:"cb.en.u4cse18042@cb.students.amrita.edu",
                password:"logins"
            }
            axios.post('https://auems2.herokuapp.com/api/signin',request)
            .then(response=>{
            console.log(response.data)
            expect(response.data.length).toEqual('valid');
        })
       
        });

        test('should test Header component', () => {
            
            expect(wrapper).toMatchSnapshot();
           });

        

        //UNIT TESTING 

        


    ;


   
  
 
  


    });

















    