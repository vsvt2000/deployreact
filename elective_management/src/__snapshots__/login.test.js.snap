// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should test Header component 1`] = `ShallowWrapper {}`;
